sodium/test/crypto_box_test.pony

```diff
@@ -38,7 +38,25 @@ class CryptoBoxTest is UnitTest
     try CryptoBox.open(crypt', nonce', csk, bpk)
       h.fail("Cyril shouldn't be able to open Bob's message to Alice.")
     end
-    
+
+    ///
+    // Key pairs generated with seeds
+
+    (let sks, let pks) = CryptoBox.seed_keypair(CryptoBoxSeed("Hello seeds!"))
```

Comment:
That's an interesting question, about being able to zero out the memory.  In Pony, the ability to mutate a value (like zeroing out its memory) is governed by the rcaps, and we are using `val` here, which doesn't ever allow mutation again (and doesn't allow anyone else to do mutation).

However, if we do a quick audit of the `crypto_*` C functions and make sure that none of them hold on to the buffer pointers past their return (off the top of my head I think they should be safe, but we'd want to make sure), then it would be possible for us to loosen the requirements to `box` instead of `val`, which would allow the caller to hold onto a `ref` (or `trn`) mutable reference if they wish, allowing them to zero out memory later.  Or they could choose to use a `val` as they do now, which also works as an argument to a `box` parameter.